### PR TITLE
MNT Use React Testing Library

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-admin/issues/1506

### CODE REVIEW

File: client/src/state/methodAvailability/tests/withMethodAvailability-test.js [hunk 1]
```diff
@@ -1,191 +0,0 @@
-/* global jest, describe, it, expect */
```
Comment (emteknetnz) [inline client/src/state/methodAvailability/tests/withMethodAvailability-test.js:1]: I simply can't find a way to test this using react testing library

File: package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
   },
   "dependencies": {
     "@silverstripe/react-injector": "^0.2.1",
+    "@testing-library/react": "^14.0.0",
```
Comment (ChloeSartorelli) [inline package.json]: Should be a dev dependency

File: client/src/components/FormField/RegisteredMFAMethodListField/AccountResetUI.js [hunk 3]
```diff
@@ -238,4 +239,8 @@ AccountResetUI.propTypes = {
   resetEndpoint: PropTypes.string,
 };
 
+AccountResetUI.defaultProps = {
+  LoadingIndicatorComponent: LoadingIndicator,
```
Comment (ChloeSartorelli) [inline client/src/components/FormField/RegisteredMFAMethodListField/AccountResetUI.js:244]: Give a proptype for this as well.

File: client/src/components/FormField/RegisteredMFAMethodListField/MethodListItem.js [hunk 4]
```diff
@@ -182,6 +182,8 @@ MethodListItem.defaultProps = {
   canRemove: false,
   canReset: false,
   tag: 'li',
+  RemoveComponent: Remove,
+  SetDefaultComponent: SetDefault
```
Comment (ChloeSartorelli) [inline client/src/components/FormField/RegisteredMFAMethodListField/MethodListItem.js:188]: Give these proptypes

File: client/src/components/FormField/RegisteredMFAMethodListField/RegisteredMFAMethodListField.js [hunk 5]
```diff
@@ -276,6 +277,8 @@ RegisteredMFAMethodListField.propTypes = {
 
 RegisteredMFAMethodListField.defaultProps = {
   initialAvailableMethods: [],
+  MethodListItemComponent: MethodListItem,
+  RegisterModalComponent: RegisterModal
```
Comment (ChloeSartorelli) [inline client/src/components/FormField/RegisteredMFAMethodListField/RegisteredMFAMethodListField.js:283]: Give these proptypes

File: client/src/components/FormField/RegisteredMFAMethodListField/tests/AccountResetUI-test.js [hunk 6]
```diff
@@ -1,151 +1,92 @@
...
-
-      ui.instance().setState({
-        complete: true,
-        failed: false,
-      });
-
-      const message = ui.find('.account-reset-action__message');
+test('AccountResetUI does not display a status by default', () => {
+  const { container } = render(<AccountResetUI {...makeProps()} />);
+  expect(container.querySelector('.account-reset-action__message')).toBeNull();
```
Comment (ChloeSartorelli) [inline client/src/components/FormField/RegisteredMFAMethodListField/tests/AccountResetUI-test.js]: Checking if this class query returns null is meaningless without _also_ checking that it _doesn't_ return null when we _do_ have a message.
The original tests do validate that (see "displays a complete status when the request succeeds" test). We need to validate it as well.

File: client/src/components/Register.js [hunk 7]
```diff
@@ -337,6 +337,10 @@ Register.defaultProps = {
   showTitle: true,
   showSubTitle: true,
   showIntroduction: true,
+  IntroductionComponent: Introduction,
+  SelectMethodComponent: SelectMethod,
+  CompleteComponent: Complete,
+  TitleComponent: Title
```
Comment (ChloeSartorelli) [inline client/src/components/Register.js:347]: Need proptypes

File: client/src/components/Register/Introduction.js [hunk 8]
```diff
@@ -131,6 +131,7 @@ Introduction.propTypes = {
 
 Introduction.defaultProps = {
   showTitle: true,
+  TitleComponent: Title
```
Comment (ChloeSartorelli) [inline client/src/components/Register/Introduction.js:135]: Need proptype

File: client/src/components/Register/SelectMethod.js [hunk 9]
```diff
@@ -142,6 +142,8 @@ SelectMethod.propTypes = {
 
 SelectMethod.defaultProps = {
   showTitle: true,
+  TitleComponent: Title,
+  MethodTileComponent: MethodTile
```
Comment (ChloeSartorelli) [inline client/src/components/Register/SelectMethod.js:148]: Need proptypes

File: client/src/components/Verify.js [hunk 10]
```diff
@@ -383,6 +383,10 @@ Verify.propTypes = {
   defaultMethod: PropTypes.string,
 };
 
+Verify.defaultProps = {
+  SelectMethodComponenet: SelectMethod
```
Comment (ChloeSartorelli) [inline client/src/components/Verify.js]: need proptype

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: client/src/components/FormField/RegisteredMFAMethodListField/AccountResetUI.js (hunk 3 above)
Comment (emteknetnz) [reply client/src/components/FormField/RegisteredMFAMethodListField/AccountResetUI.js:244]: Updated

File: client/src/components/FormField/RegisteredMFAMethodListField/MethodListItem.js (hunk 4 above)
Comment (emteknetnz) [reply client/src/components/FormField/RegisteredMFAMethodListField/MethodListItem.js:188]: Updated

File: client/src/components/FormField/RegisteredMFAMethodListField/RegisteredMFAMethodListField.js (hunk 5 above)
Comment (emteknetnz) [reply client/src/components/FormField/RegisteredMFAMethodListField/RegisteredMFAMethodListField.js:283]: Updated

File: client/src/components/FormField/RegisteredMFAMethodListField/tests/AccountResetUI-test.js (hunk 6 above)
Comment (emteknetnz) [reply client/src/components/FormField/RegisteredMFAMethodListField/tests/AccountResetUI-test.js]: Is tested in the test above 'AccountResetUI submits the reset request when clicked and hides the button' - have added an assertion to that test to check for '.account-reset-action__message'

File: client/src/components/Register.js (hunk 7 above)
Comment (emteknetnz) [reply client/src/components/Register.js:347]: Updated

File: client/src/components/Register/Introduction.js (hunk 8 above)
Comment (emteknetnz) [reply client/src/components/Register/Introduction.js:135]: Updated

File: client/src/components/Register/SelectMethod.js (hunk 9 above)
Comment (emteknetnz) [reply client/src/components/Register/SelectMethod.js:148]: Updated

File: client/src/components/Verify.js (hunk 10 above)
Comment (emteknetnz) [reply client/src/components/Verify.js]: Updated

File: package.json (hunk 2 above)
Comment (emteknetnz) [reply package.json]: Updated

**Review by GuySartorelli — APPROVED** [review]

**Pull request merged**